plugins/cloud-aws/src/main/java/org/elasticsearch/cloud/aws/AwsEc2Service.java

```diff
@@ -50,10 +50,14 @@
     @Inject
     public AwsEc2Service(Settings settings, SettingsFilter settingsFilter, NetworkService networkService, DiscoveryNodeService discoveryNodeService) {
         super(settings);
+        // Filter global settings
         settingsFilter.addFilter("cloud.key");
         settingsFilter.addFilter("cloud.account");
         settingsFilter.addFilter("cloud.aws.access_key");
         settingsFilter.addFilter("cloud.aws.secret_key");
+        // Filter repository-specific settings
+        settingsFilter.addFilter("access_key");
```

Comment:
We need both. Before filters are applied the settings are flattened, so `access_key` doesn't match `cloud.aws.access_key`. The `cloud.aws.access_key` portion filters out the keys in global settings and `access_key` filters out the key in repository settings where it appears without any prefix. I could have added `*.access_key` instead but it seems overreaching.